modules/search-pipeline-common/src/main/java/org/opensearch/search/pipeline/common/README.md

```diff
@@ -0,0 +1,119 @@
+- [Search Pipelines](#search-pipelines)
+  - [Architecture](#architecture)
+  - [Search Processors](#search-processors)
+    - [Creating a Search Processor](#creating-a-search-processor)
+    - [Creating a Pipeline](#creating-a-search-pipeline)
+
+# Search Pipelines
+
+This README briefly covers the two types of search processors, explains how you can use them to create search pipelines, and walks through the creation of a new processor.
+
+## Architecture
+
+Search pipelines allow cluster operators to create and reuse [components](#search-processors) to transform search queries and results.
+
+With search pipelines, the operator can combine multiple [search processors](#search-processors) to create a transform which acts on the search request and/or search response.
+
+The primary benefit of search pipelines is that they live entirely inside OpenSearch and therefore require less development time. Further, search pipelines support standard APIs for accessing metrics and disaster recovery.
+
+## Search Processors
+
+You can create many search pipelines by combining search processors in various orders. There are two types of search processors:
+
+1. search request processors which transform a request _before_ it is executed;
+2. search response processors which transform the output of a request _after_ it is executed.
+
+Currently, there are two search request and two search response processors.
+
+### Creating a search processor
+
+New search processors can be created by following these steps:
+
+1. Create a new class in `org.opensearch.search.pipeline.common`, this class will hold your new processor and should include whether it is a request or response processor. For example, a response processor which deleted a target field could be called `DeleteFieldResponseProcessor`.
```

Comment:
In general, you can define a processor in your own `SearchPipelinePlugin` if you don't think it's going to be useful to many OpenSearch users or if it adds some dependency on an external system that you don't want to impose on all OpenSearch users (as was the case for https://github.com/opensearch-project/search-processor/blob/main/amazon-personalize-ranking/src/main/java/org/opensearch/search/relevance/AmazonPersonalizeRankingPlugin.java). 

If someone has a processor that they would like to make available to all OpenSearch users, even in the min distribution, they should add it here (after creating an issue and getting enthusiasm from the community).

Reply:
Let me add this explanation. I will keep the example of adding the class but also note the way that the ranker was implemented.

Reply:
We may want to link to the process to develop a plugin if folks want to take that route. (It is a lot more flexible than trying to get something into core, with the trade-off of lower impact/reach.)

Reply:
You mean when mentioning the ranking processor? 

This is what I have now:

```

New search processors can be created in two different ways. 

Generally, a search processor can be created in your own `SearchPipelinePlugin`. This method is best for when you are creating a unique search
processor for your niche application. This method should also be used when your processor relies on an outside service. For an example of a 
processor which was implemented in this manner, you can reference the [`personalized_search_ranking` processor](https://github.com/opensearch-project/search-processor/blob/7e56847fa9d9e6e8201eb92d91802ec54abedcaf/amazon-personalize-ranking/src/main/java/org/opensearch/search/relevance/AmazonPersonalizeRankingPlugin.java). 

Alternatively, if you think your processor may be valuable to _all_ OpenSearch users you can follow these steps:

1. Create a new class in `org.opensearch.search.pipeline.common`, this class will hold your new processor and should include whether it is a request or response processor. For example, a response processor which deleted a target field could be called `DeleteFieldResponseProcessor`.

...
```

Reply:
Instead of recommending that folks look too closely at the Personalize ranking plugin (since it's got a bunch of code related to combining weighted scores that can be a bit confusing), maybe it's better to just point to https://github.com/opensearch-project/opensearch-plugin-template-java ?

Once you've created an empty `SearchPipelinePlugin`, adding a processor to that is essentially the same process as adding a processor to `SearchPipelineCommonModulePlugin`.

Reply:
Sure will do!